packages/core/strapi/src/core/app-configuration/config-loader.ts

```diff
@@ -9,7 +9,17 @@ export default (dir: string) => {
 
   return fs
     .readdirSync(dir, { withFileTypes: true })
-    .filter((file) => file.isFile() && VALID_EXTENSIONS.includes(path.extname(file.name)))
+    .filter((file) =>{
+      const fullPath = path.join(dir, file.name);
+      let stat: fs.Stats | fs.Dirent;
+      try {
+        stat = file.isSymbolicLink() ? fs.statSync(fullPath) : file;
+      } catch(err) {
+        // ignore broken symbolic link
+        return false;
```

Comment:
We might want to still throw the error the same as if we attempt to load a .js file that has errors and can't be loaded.

Thoughts?

Reply:
Good idea, I was too tunnel visioned on keeping the original behavior 😅. I've added an error throw as well as adapted the tests for the the change.